lang/rust/avro/src/reader.rs

```diff
@@ -396,9 +396,9 @@ impl GenericSingleObjectReader {
 
     pub fn read_value<R: Read>(&self, reader: &mut R) -> AvroResult<Value> {
         let mut header: [u8; 10] = [0; 10];
-        match reader.read(&mut header) {
-            Ok(size) => {
-                if size == 10 && self.expected_header == header {
+        match reader.read_exact(&mut header) {
+            Ok(_) => {
```

Comment:
What would you feel about pattern matching to 10?
Along with a follow up error for more details, something like:
```
Ok(10) => { //this case},
Ok(x) => Err(Error::InompleteHeader(x)),
Err(io_error)=> Err(Error::ReadHeader(io_error)), //existing below
```

Reply:
`reader.read_exact` returns `Result<()>`, so it is not possible to match to 10 here.

Reply:
That would make sense. LGTM